src/scope-locals.ts

```diff
@@ -12,6 +12,45 @@ import { ASTPluginEnvironment, NodeVisitor } from '@glimmer/syntax';
 import { astNodeHasBinding } from './hbs-utils';
 import { readOnlyArray } from './read-only-array';
 
+/**
+ * RFC: Pending....
+ *
+ * tl;dr:
+ *   - utilities that begin with uppercase letter
+ *     OR are guaranteed to never be added to glimmer as a keyword (e.g.: globalThis)
+ *   - must not need `new` to invoke
+ *
+ * Ref:
+ * - https://tc39.es/ecma262/#sec-global-object
+ * - https://tc39.es/ecma262/#sec-function-properties-of-the-global-object
+ */
+const ALLOWED_GLOBALS = new Set([
+  // namespaces
+  'globalThis',
+  'JSON',
+  'Math',
+  'Atomics',
+  'Reflect',
+  // functions
+  'isNaN',
+  'isFinite',
+  'parseInt',
+  'parseFloat',
+  'decodeURI',
+  'decodeURIComponent',
+  'encodeURI',
+  'encodeURIComponent',
+  // new-less Constructors (still functions)
+  'Number',
+  'Object', // different behavior from (hash)
+  'Array', // different behavior from (array)
+  'String',
+  'BigInt',
+  'Date',
```

Comment:
I think `Boolean` meets the criteria... handy for turning something truthy to meet an arg typed as boolean only

Reply:
That's a good point. How did we miss that one? Are we reviewing all the right parts of the spec?

Reply:
resolved (it is in the RFC)